pkg/controller/jobframework/setup_test.go

```diff
@@ -135,18 +158,44 @@ func TestSetupControllers(t *testing.T) {
 				t.Fatalf("Failed to setup manager: %v", err)
 			}
 
-			gotError := manager.setupControllers(mgr, logger, tc.opts...)
+			gotError := manager.setupControllers(context.Background(), mgr, logger, time.Millisecond*20, tc.opts...)
 			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); len(diff) != 0 {
 				t.Errorf("Unexpected error from SetupControllers (-want,+got):\n%s", diff)
 			}
 
-			if diff := cmp.Diff(tc.wantEnabledIntegrations, manager.enabledIntegrations.SortedList()); len(diff) != 0 {
+			if tc.afterSetup != nil {
+				tc.afterSetup(mgr, manager)
+			}
+
+			diff := cmp.Diff(tc.wantEnabledIntegrations, getEnabledIntegrations(manager).SortedList())
+			if len(diff) != 0 {
 				t.Errorf("Unexpected enabled integrations (-want,+got):\n%s", diff)
 			}
 		})
 	}
 }
 
+func testDelayedIntegration(mgr ctrlmgr.Manager, manager *integrationManager) {
+	rayGVK := schema.GroupVersionKind{Group: "ray.io", Version: "v1", Kind: "RayCluster"}
+	manager.mu.Lock()
+	mgr.GetRESTMapper().(*apimeta.DefaultRESTMapper).Add(rayGVK, apimeta.RESTScopeNamespace)
+	manager.mu.Unlock()
+	// Wait for setup to complete
+	for {
+		_, ok := getEnabledIntegrations(manager)["ray.io/raycluster"]
+		if ok {
+			break // Exit loop if RayCluster is enabled
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+}
+
+func getEnabledIntegrations(manager *integrationManager) set.Set[string] {
```

Comment:
No, I meant a method of `integrationManager`. You can use it inside `integrationManager` to do not duplicate every time lock/unlock.

Reply:
Ohh I understand now, please see latest changes if it meets with expectations. Thanks for the help